### PR TITLE
cleanup voting registration

### PR DESCRIPTION
**Pull request by wolverineks**
~~this pr requires removing `signer` from creating voting registration tx~~

### CODE REVIEW

File: src/legacy/fetch.ts [hunk 1]
```diff
@@ -45,11 +45,17 @@ export const checkedFetch = (request: FetchRequest) => {
   const {endpoint, payload, method, headers} = request
   const checkResponse = request.checkResponse || _checkResponse
   Logger.info(`API call: ${endpoint}`)
-  return fetch(endpoint, {
-    method,
-    headers: headers != null ? headers : undefined,
-    body: payload != null ? JSON.stringify(payload) : undefined,
-  }) // Fetch throws only for network/dns/related errors, not http statuses
+
+  const args = [
```
Comment (wolverineks) [inline src/legacy/fetch.ts:49]: makes copying args easier in debugger

File: src/TxHistory/BalanceBanner.tsx [hunk 2]
```diff
@@ -27,14 +27,15 @@ export const BalanceBanner = () => {
       <Spacer height={10} />
 
       <TouchableOpacity onPress={() => setPrivacyMode(!privacyMode)} style={styles.button}>
-        <Row>
```
Comment (wolverineks) [inline src/TxHistory/BalanceBanner.tsx:30]: eliminates large loading when logging in

File: src/Menu/Menu.tsx [hunk 3]
```diff
@@ -19,14 +19,15 @@ const MenuStack = createStackNavigator()
 
 export const MenuNavigator = () => {
   const strings = useStrings()
+  const wallet = useSelectedWallet()
+  usePrefetchStakingInfo(wallet)
 
   return (
     <MenuStack.Navigator
-      initialRouteName="menu"
+      initialRouteName="_menu"
```
Comment (wolverineks) [inline src/Menu/Menu.tsx:27]: duplicate route names

File: src/Menu/Menu.tsx [hunk 4]
```diff
@@ -19,14 +19,15 @@ const MenuStack = createStackNavigator()
...
       screenOptions={{...defaultStackNavigationOptions, headerLeft: () => null}}
     >
-      <MenuStack.Screen name="menu" component={Menu} options={{title: strings.menu}} />
-      <MenuStack.Screen name="catalyst-voting" component={CatalystNavigator} />
```
Comment (wolverineks) [inline src/Menu/Menu.tsx:29]: never used

File: src/Menu/Menu.tsx (hunk 3 above)
Comment (wolverineks) [inline src/Menu/Menu.tsx:27]: possibly, remove this navigator altogether

File: src/Dashboard/DashboardNavigator.tsx [hunk 5]
```diff
@@ -68,5 +67,5 @@ const messages = defineMessages({
 const HeaderRight = () => {
   const {navigateToSettings} = useWalletNavigation()
 
-  return <SettingsButton onPress={() => navigateToSettings()} />
+  return <SettingsButton style={{paddingRight: 16}} onPress={() => navigateToSettings()} />
```
Comment (wolverineks) [inline src/Dashboard/DashboardNavigator.tsx:70]: fixes header spacing

File: src/Catalyst/hooks.ts [hunk 6]
```diff
@@ -1,54 +1,10 @@
...
-
-export type VotingRegTxVariables = {
-  pin: string
-  decryptedKey?: string
-}
-export type VotingRegTxData = {
-  catalystSKHexEncrypted: string
-  yoroiUnsignedTx: YoroiUnsignedTx
-}
-export const useCreateVotingRegTx = (
```
Comment (wolverineks) [inline src/Catalyst/hooks.ts:18]: moved to hooks/index
useMutation -> useQuery

**Review by stackchain — APPROVED** [review]

**Review by banklesss — APPROVED** [review]

**Pull request merged**